src/lib/y2storage/clients/inst_disk_proposal.rb

```diff
@@ -155,6 +165,35 @@ def new_proposal(settings)
         probed = storage_manager.probed
         GuidedProposal.new(settings: settings, devicegraph: probed, disk_analyzer: probed_analyzer)
       end
+
+      # Buttons to be excluded in the proposal dialog
+      #
+      # @return [Array<Symbol>]
+      def excluded_buttons
+        excluded = []
+        excluded << :guided unless show_guided_setup?
+        excluded
+      end
+
+      # Whether it is possible to use the Guided Setup
+      #
+      # @note Even when the "proposal_settings_editable" from control file is set to not allow
+      #   to edit the proposal settings, the button for the Guided Setup can be still offered
+      #   to allow to select in which disks to install (two first steps of the Guided Setup).
+      #
+      # @see Dialogs::GuidedSetup.allowed?
+      #
+      # @return [Boolean]
+      def show_guided_setup?
+        Dialogs::GuidedSetup.allowed? || several_candidate_disks?
+      end
+
+      # Whether there are more than one candidate disks
+      #
+      # @return [Boolean]
+      def several_candidate_disks?
```

Comment:
Not sure why the logic about several_candidate_disks? is here and not in the `GuidedSetup` class. That's kind of splitting the logic into two places.